react-first-app/src/app/components/App/Game/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+
+import logo from '../../../../logo.svg';
+
+import style from './styles.css';
+
+import Board from '../Board';
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null)
+        }
+      ],
+      stepNumber: 0,
+      xIsNext: true
+    };
+  }
+
+  handleClick(i) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? "X" : "O";
+    this.setState({
+      history: history.concat([
+        {
+          squares: squares
+        }
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    });
+  }
+
+  jumpTo(step) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: (step % 2) === 0
+    });
+  }
+
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+
+    const moves = history.map((step, move) => {
```

Comment:
We try to avoid logic in render(), so as to have a better visibility of what's being rendered. I'd extract logic outside the render so all wee see is JSX when possible:

```
  render() {
    const history = this.state.history;
    const current = history[this.state.stepNumber];
  
    return (
      <div className="game">
        <div className="game-board">
          <Board squares={current.squares} onClick={i => this.handleClick(i)} />
        </div>
        <div className="game-info">
          {this.getStatus()}
          <ol>
            {history.map((step, move) => (
              <li key={move}>
                <button onClick={() => this.jumpTo(move)}>{this.getDesc(move)}</button>
              </li>
            ))}
          </ol>
        </div>
      </div>
    );
  }
```